tests/test_abp.py

```diff
@@ -51,8 +51,9 @@
 
 @pytest.mark.slow
 @pytest.mark.use_python
+@pytest.mark.parametrize('num_threads', [1, 4])
```

Comment:
Lets not hardcode this incase there are machines with less cores. Can we use `os.cpu_count()` instead of 4?